pvlib/pvsystem.py

```diff
@@ -2036,9 +2037,20 @@ def calcparams_desoto(effective_irradiance, temp_cell,
     # use errstate to silence divide by warning
     with np.errstate(divide='ignore'):
         Rsh = R_sh_ref * (irrad_ref / effective_irradiance)
+
     Rs = R_s
 
-    return IL, I0, Rs, Rsh, nNsVth
+    numeric_args = (effective_irradiance, temp_cell)
+    out = (IL, I0, Rs, Rsh, nNsVth)
+
+    if all(map(np.isscalar, numeric_args)):
```

Comment:
With the goal of `calcparams_*` methods' return values having the same Python type, I think we want the behavior of `np.isscalar` instead of `np.ndim(x) == 0`. If `calcparams_*` is given a mix of floats and 0d-arrays (e.g. `effective_irradiance` is a Python `float` and `temp_cell` is a 0d-array), using `np.ndim(x) == 0` will cause the `calcparams_*` methods' return values to be a mix of floats and 0d-arrays. If we use `np.isscalar`, all return values will become 0d-arrays. This is because `np.broadcast_arrays(1.0, np.array(2.0))` equals `[np.array(1.0), np.array(2.0)]`.